Adds the aws_lambda_permission resource to the Terraform example

bwhaley opened the pull request:
<!-- *Note: Please remember to review the Datadog Documentation [Contribution Guidelines](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md) if you have not yet done so.* -->

### What does this PR do?
Adds an example of the `aws_lambda_permission` Terraform resource to the manual triggers documentation.

### Motivation
Log forwarding will not work without the `aws_lambda_permission` in place.

<!-- ### Preview -->
<!-- Assuming you are a Datadog employee and named your branch `<yourname>/<description>`, a preview build will run and links to the preview output will be auto-generated and posted in the PR comments. The links will 404 until the preview build is finished running -->


---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

maycmlee commented:
Thanks @bwhaley for opening this PR! We're reviewing it and will be back in touch.

buraizu commented:
Thanks again @bwhaley for the PR. Could you confirm the steps taken and the error received when you tried to set this up without the `aws_lambda_permission` resource? I wasn't able to replicate this in testing with a sandbox Datadog account -- the subscription filter was successfully created and logs forwarded from a CloudWatch logs group using the example Terraform block in our documentation, without creating any additional permissions. 
Thanks in advance for any additional details -- this will help us make sure the docs get fully clarified.

bwhaley commented:
I actually couldn't find an error, but the logs weren't ever being forwarded to Datadog until I added the permission. I noticed that some of our other log groups were added using the console and they did have permissions associated with them. That's why I thought to add a permission, and sure enough, when I added it, the logs started forwarding correctly.

AWS provider v4.28.0.

buraizu commented:
Thanks @bwhaley for those additional details. The log group I was first using was created through the AWS console, so I tried a fresh example with a log group created through the AWS CLI and still was able to get logs in Datadog by running the Terraform block without creating additional permissions. I'll reach out internally to look further into what might be the cause of this, and follow up with any updates -- thank you for your patience!

buraizu commented:
Hi @bwhaley, thanks for your patience here. After checking with the team, we haven’t been able to replicate this issue. If you encounter this issue again, could you open a [support ticket](https://docs.datadoghq.com/help/) so our support team can investigate what might be going on? They may have additional insights into the requirements here as well as knowledge about possible edge cases.
I’ll close this PR for now, but please feel free to reopen if you’re able to confirm this requirement.
